identity/src/main/java/com/android/identity/credential/CredentialUtil.kt

```diff
@@ -0,0 +1,130 @@
+     * from the issuer.
+     *
+     * @param credential the credential to manage authentication keys for.
+     * @param secureArea the secure area to use for new pending authentication keys.
+     * @param createKeySettings the settings used to create new pending authentication keys.
+     * @param domain the domain to use for created authentication keys.
+     * @param now the time right now, used for determining which existing keys to replace.
+     * @param numAuthenticationKeys the number of authentication keys that should be kept.
+     * @param maxUsesPerKey the maximum number of uses per key.
+     * @param minValidTimeMillis requests a replacement for a key if it expires within this window.
+     * @param dryRun don't actually create the keys, just return how many would be created.
+     * @return the number of pending authentication keys created.
+     */
+    @JvmStatic
+    fun managedAuthenticationKeyHelper(
+        credential: Credential,
+        secureArea: SecureArea,
+        createKeySettings: CreateKeySettings?,
+        domain: String,
+        now: Timestamp,
+        numAuthenticationKeys: Int,
+        maxUsesPerKey: Int,
+        minValidTimeMillis: Long,
+        dryRun: Boolean
+    ): Int {
+        // First determine which of the existing keys need a replacement...
+        var numKeysNotNeedingReplacement = 0
+        var numReplacementsGenerated = 0
+        for (authKey in credential.authenticationKeys) {
+            var keyExceededUseCount = false
+            var keyBeyondExpirationDate = false
+            if (authKey.domain != domain) {
+                // not one of ours...
+                continue
+            }
+            if (authKey.usageCount >= maxUsesPerKey) {
+                keyExceededUseCount = true
+            }
+            val expirationDate = Timestamp.ofEpochMilli(
+                authKey.validUntil.toEpochMilli() - minValidTimeMillis
+            )
+            if (now.toEpochMilli() > expirationDate.toEpochMilli()) {
+                keyBeyondExpirationDate = true
+            }
+            if (keyExceededUseCount || keyBeyondExpirationDate) {
+                if (authKey.replacement == null) {
+                    if (!dryRun) {
+                        credential.createPendingAuthenticationKey(
+                            domain,
+                            secureArea,
+                            createKeySettings!!,
+                            authKey
+                        )
+                    }
+                    numReplacementsGenerated++
+                    continue
+                }
+            }
+            numKeysNotNeedingReplacement++
+        }
+
+        var numExistingPendingKeys = 0
+        for (pendingAuthenticationKey in credential.pendingAuthenticationKeys) {
+            if (pendingAuthenticationKey.domain == domain) {
+                numExistingPendingKeys += 1
+            }
+        }
```

Comment:
We can just count the results after applying `filter` in the new for-loop above (which infers that the variable be extracted outside of the for-loop so we can get the "numExistingPendingKeys"

```
val numExistingPendingKeys = existingPendingKeys.size
```